packages/core/crates/core-mixer/src/future_extensions.rs

```diff
@@ -0,0 +1,91 @@
+//! Highly inspired by [and_then_concurrent](https://docs.rs/and-then-concurrent/latest/and_then_concurrent/)
+//!
+//! `ThenConcurrent` extension adds the `then_concurrent` method to any `futures::stream::Stream`
+//! object allowing a concurrent execution of futures over the stream items.
+
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use futures::stream::{FuturesUnordered, Stream};
+use pin_project::pin_project;
+
+/// Stream for the [`Stream::then_concurrent()`] method.
+#[pin_project(project = ThenConcurrentProj)]
```

Comment:
why don't use
```rust
pin_project! {
   struct { ... }
}
```

Reply:
Just the basic version the library used in their documentation.